dlrs/main/classes/RollupSummariesSelector.cls

```diff
@@ -90,41 +113,70 @@ public class RollupSummariesSelector {
     List<RollupSummaries.CalculationMode> calculationModes,
     Set<String> childObjectNames
   ) {
+    List<String> rollupSummaryNames = new List<String>();
+    for (RollupSummaries.CalculationMode cm : calculationModes) {
+      rollupSummaryNames.add(cm.name());
+    }
     List<RollupSummary> records = new List<RollupSummary>();
-    records = m_customObjectSelector.selectActiveByChildObject(
-      calculationModes,
-      childObjectNames
-    );
-    records.addAll(
-      m_customMetadataSelector.selectActiveByChildObject(
-        calculationModes,
-        childObjectNames
-      )
-    );
-    return records;
+    for (RollupSummary rs : this.allSummaries) {
+      if (
+        rs.Active &&
+        containsIgnoreCase(childObjectNames, rs.ChildObject) &&
+        containsIgnoreCase(rollupSummaryNames, rs.CalculationMode)
+      ) {
+        records.add(rs);
+      }
+    }
+    return sortSummaries(records, 'ParentObject__c', 'RelationshipField__c');
   }
 
   /**
    * Returns active lookup rollup summary definitions for the given rollup unique names
    **/
   public List<RollupSummary> selectActiveByUniqueName(Set<String> uniqueNames) {
     List<RollupSummary> records = new List<RollupSummary>();
-    records = m_customObjectSelector.selectActiveByUniqueName(uniqueNames);
-    records.addAll(
-      m_customMetadataSelector.selectActiveByUniqueName(uniqueNames)
-    );
+    for (RollupSummary rs : selectByUniqueName(uniqueNames)) {
+      if (rs.Active) {
+        records.add(rs);
+      }
+    }
     return records;
   }
 
+  /**
+   * Returns active lookup rollup summary definitions for the given rollup unique names
+   **/
+  public List<RollupSummary> selectByUniqueName(Set<String> uniqueNames) {
+    List<RollupSummary> records = new List<RollupSummary>();
+    for (RollupSummary rs : this.allSummaries) {
+      if (containsIgnoreCase(uniqueNames, rs.UniqueName)) {
+        records.add(rs);
+      }
+    }
+    return sortSummaries(records, 'ParentObject__c', 'RelationshipField__c');
+  }
+
   /**
    * Returns lookup rollup summary definitions for all fields
    **/
   public List<RollupSummary> selectAll() {
-    List<RollupSummary> records = new List<RollupSummary>();
-    records = m_customMetadataSelector.selectAll();
-    records.addAll(m_customObjectSelector.selectAll());
+    return this.allSummaries;
+  }
 
-    return records;
+  private Boolean containsIgnoreCase(List<String> collection, String val) {
+    if (val == null) {
+      return collection.contains(val);
+    }
+    for (String s : collection) {
+      if (val.equalsIgnoreCase(s)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private Boolean containsIgnoreCase(Set<String> collection, String val) {
```

Comment:
with api 58 ( see your other PR) set implements iterable

Reply:
I think that would be a great way to do it. Just take a param of iterable string and not need the overrides. I expect the 58 to be merged first then I can modify this PR before it is merged. Thanks!